core/src/test/scala/unit/kafka/log/LogTest.scala

```diff
@@ -2885,6 +2813,100 @@ class LogTest {
     recoveredLog.close()
   }
 
+  private def testProducerSnapshotsRecoveryAfterUncleanShutdown(messageFormatVersion: String): Unit = {
```

Comment:
nit: probably could have left this where it was. That would make it easier to tell what has changed.